### PR TITLE
Add .coderabbit.yaml for sriov-network-device-plugin

### PR DESCRIPTION
**Pull request by schsebabot**
## Summary

Adds a `.coderabbit.yaml` configuration file to enable and configure CodeRabbit AI-powered code reviews for the sriov-network-device-plugin repository. The configuration is adapted from the [sriov-network-operator's `.coderabbit.yaml`](https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/1095), tailored to the device plugin codebase structure and concerns.

## Key Changes

- **New `.coderabbit.yaml` file** with comprehensive CodeRabbit configuration including:
  - Review profile set to "chill" with auto-review enabled on all PRs
  - Free tier enabled for open-source usage
  - Path-based review instructions covering:
    - **Go source code** (`pkg/**/*.go`): General Go best practices, error handling, concurrency safety, and Kubernetes API conventions
    - **Device plugin core** (`pkg/resources/*.go`): SR-IOV device plugin API compliance, resource pool management, device health monitoring, and gRPC lifecycle
    - **Device discovery** (`pkg/devices/*.go`, `pkg/netdevice/*.go`, `pkg/pcidevice/*.go`): PCI/netlink device enumeration, sysfs parsing, and NUMA-aware allocation
    - **VF allocation** (`pkg/resources/pool*.go`): Virtual Function allocation/deallocation, device ID tracking, and kubelet checkpoint consistency
    - **Selectors & filtering** (`pkg/types/*.go`): Device selector logic, vendor/device/driver filtering, and link type matching
    - **Unit tests** (`pkg/**/*_test.go`): Table-driven tests, mock usage, and edge case coverage
    - **E2E tests** (`test/**`): Kubernetes cluster dependencies, SR-IOV hardware requirements, and CI reproducibility
    - **Dockerfiles** (`images/**/Dockerfile*`): Multi-stage builds, base image pinning, and non-root execution
    - **Go module management** (`go.mod`, `go.sum`): Supply chain security and dependency review
    - **CI/CD workflows** (`.github/workflows/**`): GitHub Actions security best practices
    - **Kubernetes manifests** (`deployments/*.yaml`): RBAC least privilege, resource limits, and node affinity for SR-IOV capable nodes
    - **Security/Cryptography** (`**/*.go`): Flagging of deprecated or weak cryptographic usage
  - Path filters to reduce noise on auto-generated and vendor files
  - Knowledge base integration with learnings and issue tracking enabled

## Notable Implementation Decisions

- **Operator-specific instructions removed**: Controllers, webhooks, bindata manifests, and other operator-pattern instructions from the reference config were intentionally excluded as they don't apply to this device plugin project
- **Device-plugin-specific paths added**: New path instruction groups for device discovery (`pkg/devices/`, `pkg/netdevice/`, `pkg/pcidevice/`), VF allocation (`pkg/resources/pool*.go`), and selector/filtering logic (`pkg/types/`) reflect the unique architecture of the device plugin
- **Same review profile and security posture**: Maintains the "chill" review profile, security scanner configuration, and knowledge base settings from the reference sriov-network-operator configuration for consistency across the k8snetworkplumbingwg organization

## Summary by CodeRabbit

* **Chores**
  * Introduced a new CodeRabbit configuration to standardize automated reviews and enable security-focused scanning.
  * Enabled draft review generation and set review behavior preferences, including language and review profile.
  * Added path-based review guidance and extensive security/best-practice rules across Go code, Kubernetes manifests, CI/CD, Docker, and cryptography.

### CODE REVIEW

File: .coderabbit.yaml [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        - DeviceType enum: ensure new device types are added to all switch
+          statements and factory methods
+        - Mock generation: changes to interfaces require regenerating mocks
+          (make generate-mocks) — flag if interface changed but mocks were not updated
+        - Exported types: verify JSON tags, documentation, and field validation
+        - VdpaType: ensure virtio and vhost types are handled consistently
+
+    # ── Kubernetes manifests ─────────────────────────────────────
+    - path: "**/*.{yaml,yml}"
+      instructions: |
```
[the remark this answers is not in the document]
Comment (adrianchiris) [reply .coderabbit.yaml]: seems valid, maybe limit this to the deployment yaml dir

[the remark this answers is not in the document]
Comment (schsebabot) [reply .coderabbit.yaml]: Good call, done — I've scoped it to `deployments/**/*.{yaml,yml}` and `images/**/*.{yaml,yml}` so the Kubernetes hardening instructions only fire on actual manifest directories. Also removed the `go.sum` exclusion from `path_filters` as suggested by the other comment, since we want visibility into module checksum changes for supply-chain review.

**Review by zeeke — APPROVED** [review]

**Review by adrianchiris — APPROVED** [review]
